sdk/python/kfp/components/_structures.py

```diff
@@ -165,13 +168,63 @@ def __init__(self,
         super().__init__(locals())
 
 
+class InputMetadataPlaceholder(ModelBase):  # Non-standard attr names
+    """Represents the file path to an input artifact metadata.
+
+    During runtime, this command-line argument placeholder will be replaced
+    by the path where the metadata file associated with this artifact has been
+    written to. Currently only supported in v2 components.
+    """
+    _serialized_names = {
+        'input_name': 'inputMetadata',
+    }
+
+    def __init__(self, input_name: str):
+        super().__init__(locals())
+
+
+class InputOutputPortNamePlaceholder(ModelBase):  # Non-standard attr names
+    """Represents the output port name of an input artifact.
+
+    During compile time, this command-line argument placeholder will be replaced
+    by the actual output port name used by the producer task. Currently only
```

Comment:
Are you sure that the pros outweigh the cons?
This can reduce component isolation, reproducibility and affect caching.

Perhaps we should avoid giving downstream tasks information about upstream tasks other that the information produced using outputs.

Reply:
What is the high-level problem this is trying to solve?
I have a feeling that the problem can be better solved on another architecture level - DSL, compiler or execution engine.